File: problems/0009_palindrome-number/to-string/Solution.scala
object Solution {
    def isPalindrome(x: Int): Boolean = {
        if (x < 0)
            return false
        else if (x < 10)
            return true
        val s = x.toString
        var (i, j) = (0, s.length - 1)
        while (i < j) {
            if (s(i) != s(j))
                return false
            i += 1
            j -= 1
        }
        true
    }
}

File: problems/0009_palindrome-number/to-string/Solution.cpp
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        const string s = to_string(x);
        for (int i = 0, j = s.length() - 1; i < j; ++i, --j)
            if (s[i] != s[j])
                return false;
        return true;
    }
};
